internal/comparetest/golden/golden_test.go

```diff
@@ -33,7 +33,14 @@ func TestWriteMetrics(t *testing.T) {
 	metricslice.CopyTo(metrics.ResourceMetrics().AppendEmpty().ScopeMetrics().AppendEmpty().Metrics())
 
 	actualFile := filepath.Join(t.TempDir(), "metrics.json")
-	require.NoError(t, WriteMetrics(actualFile, metrics))
+
+	// Overwrite the callback, to aviod an actual failure.
+	var failed bool
+	writeMetricsCallback = func(t *testing.T) {
+		failed = true
+	}
```

Comment:
Could it affect other test cases?

Reply:
Good catch. `TestRoundTrip` fails if run independently.

I'm not too worried about problems outside of this small package because there should not be any permanent calls to `WriteMetrics`. I will set and unset the callback in both internal cases though.

The other option I see is to get rid of the callback altogether and call an unexported function for the actual writing. This leaves the exported wrapper untested though.

```go
// WriteMetrics writes a pmetric.Metrics to the specified file
func WriteMetrics(t *testing.T, filePath string, metrics pmetric.Metrics) error {
	if err := writeMetrics(filePath, metrics); err != nil {
		return err
	}
	t.Logf("Golden file successfully written to %s.", filePath)
	t.Log("NOTE: The WriteMetrics call must be removed in order to pass the test.")
	t.Fail()
	return nil
}

// writeMetrics writes a pmetric.Metrics to the specified file
func writeMetrics(filePath string, metrics pmetric.Metrics) error {
	unmarshaler := &pmetric.JSONMarshaler{}
	fileBytes, err := unmarshaler.MarshalMetrics(metrics)
	if err != nil {
		return err
	}
	var jsonVal map[string]interface{}
	if err = json.Unmarshal(fileBytes, &jsonVal); err != nil {
		return err
	}
	b, err := json.MarshalIndent(jsonVal, "", "   ")
	if err != nil {
		return err
	}
	b = append(b, []byte("\n")...)
	if err := os.WriteFile(filePath, b, 0600); err != nil {
		return err
	}
	return nil
}
```

Reply:
I think this is a better alternative. I don't think it's critical to cover `WriteMetrics`

Reply:
Ok, updated